polynote-kernel/src/main/scala/polynote/kernel/interpreter/CellExecutor.scala

```diff
@@ -40,4 +41,36 @@ class CellExecutor(publishSync: Result => Unit, classLoader: ClassLoader, blocki
     }
   }
   def here: Boolean = blockingExecutor.here
+}
+
+object CellExecutor {
+
+  // Make sure Java's console uses the thread-local mechanism of the Scala console
+  // This way it can reset properly but still obey the Console.withOut mechanism
+  lazy val initJavaConsole: Unit = {
```

Comment:
So is this just a better way to do what we were doing by using `System.setOut` or is there a difference?

Reply:
I mean, was there a bug being fixed with this change? 😄

Reply:
The difference is that we were relying on the thread doing `Blocking` to be the same thread doing the outputtin'. It's not necessarily the case anymore, so Java needs to do what Scala does and have what amounts to a `ThreadLocal` console.

Reply:
oh i see, hm. where's that change?

I wonder if this solves this issue? https://github.com/polynote/polynote/issues/679
which admittedly is poorly-written (man, who filed that?!). Basically jvm stdout (like a scala function that prints) wasn't being captured if you called it in a python cell.